src/core/Track.cpp

```diff
@@ -826,22 +830,50 @@ void TrackContentObjectView::mousePressEvent( QMouseEvent * me )
 	{
 		if( me->modifiers() & Qt::ControlModifier )
 		{
-			m_tco->toggleMute();
+			if( so.size() > 0 )
+			{
+				toggleMuteSelection( so );
+			}
```

Comment:
I think a cleaner implementation than doing all these checks is to use one function for both selections and single clips. Simply pass a collection containing a single clip if you only want to affect that clip.

I'm also not sure that this logic holds. What if I select a bunch of clips, and the press mute on an unselected clip? `QVector so` will still contain the selected objects, so `so.size > 0` should be true, and the selection will be muted instead of the clip that was clicked on.

Reply:
Nice catch, you're right! I have to check both if there are selected TCO views and if the one where the event happened is among them. I'll fix that on all actions.

I thought about having `so.size > 0` inside the `toggleMuteSelection` method, but I thought the name wouldn't describe the behavior pretty well anymore, because `toggleMuteSelection` would both mute selections or individual tracks, and I don't know, it could be useful in the future to have those methods that take selections and modify them. What do you think about making a third method that would check if `so.size > 0` and if the event TCO view is selected them and then call either `toggleMuteSelection` or `toggleMute`? What could I call it?

Reply:
I fixed the bug and added the conditionals inside the `removeSelection()`,`copySelection()`,`cutSelection()`,`toggleMuteSelection()` methods. Still unsure about the naming, but had no better ideas until now :thinking:

Reply:
I fixed the bug and added the conditionals inside the `removeSelection()`,`copySelection()`,`cutSelection()` and `toggleMuteSelection()` methods. Still unsure about the naming, but had no better ideas until now :thinking: